sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/FieldBase.java

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.documents.models;
+
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.Objects;
+
+/**
+ * Base field type for helper classes to more easily create a {@link Index}.
+ */
+public abstract class FieldBase {
```

Comment:
When you do the renames, you might call this `SearchFieldBase` given our discussion internally.

Reply:
Will do. Thanks for reminding! Will add you as review for renaming changes.